From 4fc23be1932bec3a527bb594b731a432530d4316 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 10 Jan 2024 16:53:52 +0100
Subject: [PATCH 450/841] fix(stm32mp2): properly check err values

This is required by MISRA to have boolean types for if tests.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9a5fd2b6d83463e4208ab90d1ca27b3a8ea4db9c fix(stm32mp2): properly check err values]
Change-Id: Ie778f0665d9a9cc9a58b4316ecbdc8f4a7510c0e
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/350517
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index b061e39e8..64956c0f0 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2023-2024, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -549,7 +549,7 @@ int bl2_plat_handle_post_image_load(unsigned int image_id)
 #ifndef DECRYPTION_SUPPORT_none
 		/* Load encryption key info before DT is unmapped */
 		err = stm32_get_enc_key_otp_idx_len(&otp_idx, &otp_len);
-		if (err) {
+		if (err != 0) {
 			panic();
 		}
 #endif
@@ -568,7 +568,7 @@ int bl2_plat_handle_post_image_load(unsigned int image_id)
 			err = parse_optee_header(&bl_mem_params->ep_info,
 						 &pager_mem_params->image_info,
 						 NULL);
-			if (err) {
+			if (err != 0) {
 				ERROR("OPTEE header parse error.\n");
 				panic();
 			}
-- 
2.47.2

